src/daemon/ca/certauth.rs

```diff
@@ -964,6 +973,123 @@ impl CertAuth {
         Ok(res)
     }
 
+    // Suspend a child. The intention is that this is called when it is discovered
+    // that the child has been inactive, i.e. not contacting this parent for a pro-longed
+    // period of time (hours).
+    //
+    // When a child is suspended we need to:
+    // - mark it as suspended
+    // - withdraw all certificates issued to it (suspend them)
+    fn child_suspend_inactive(&self, child_handle: &ChildHandle) -> KrillResult<Vec<CaEvt>> {
+        let mut res = vec![];
+
+        let child = self.get_child(&child_handle)?;
+
+        if child.is_suspended() {
+            return Ok(res); // nothing to do, child is already suspended
+        }
+
+        let mut version = self.version;
+        let handle = &self.handle;
+
+        // Find all the certs in all RCs for this child and suspend them.
+        for (rcn, rc) in self.resources.iter() {
+            let certified_keys = child.issued(rcn);
+
+            if certified_keys.is_empty() {
+                continue;
+            }
+
+            let mut cert_updates = ChildCertificateUpdates::default();
+
+            for key in certified_keys {
+                if let Some(issued) = rc.issued(&key) {
+                    cert_updates.suspend(issued.clone());
+                }
+            }
+
+            res.push(CaEvtDet::child_certificates_updated(
+                handle,
+                version,
+                rcn.clone(),
+                cert_updates,
+            ));
+            version += 1;
+        }
+
+        info!("CA '{}' suspended inactive child '{}'", handle, child_handle);
```

Comment:
There is no logic in this function to determine "inactivity" - perhaps remove "inactive" from this text?